### PR TITLE
Alola evolutions update

### PR DESCRIPTION
**Pull request by amative1**
Per #1003, updating Charjabug -> Vikavolt (as location-sensitive), Crabrawler -> Crabominable (as location-sensitive), Steenee -> Tsareena (as level-sensitive), and Poipole -> Naganadel (as level-sensitive).

Still pending is Meltan -> Melmetal, though this is not Alola-launch critical.

### CODE REVIEW

File: src/scripts/pokemons/PokemonList.ts [hunk 1]
```diff
@@ -15865,7 +15865,10 @@ const pokemonList = createPokemonArray(
         'levelType': LevelType.mediumfast,
         'exp': 140,
         'catchRate': 120,
-        'evolutions': [new StoneEvolution('Charjabug', 'Vikavolt', GameConstants.StoneType.Thunder_stone)],
+        'evolutions': [
+            new StoneEvolution('Charjabug', 'Vikavolt', GameConstants.StoneType.Thunder_stone),
+            new DungeonRestrictedLevelEvolution('Vast Poni Canyon','Charjabug','Vikavolt', 20),
```
Comment (RedSparr0w) [inline src/scripts/pokemons/PokemonList.ts:15870]: So `Charjabug` can be evolved via either method?

Comment (amative1) [reply src/scripts/pokemons/PokemonList.ts:15870]: That's the idea. The stone is a Gen 8 method that was already there, I just added the Gen 7 "exposure to a special magnetic field" rule.

Comment (Ultima1990) [reply src/scripts/pokemons/PokemonList.ts:15870]: I don't agree with having stone evolution as an option for Charjabug.

Comment (Askellown) [reply src/scripts/pokemons/PokemonList.ts:15870]: While dex entry is restricted by region, I agree it should be like Leafeon and restricted to area, but when it stops being restricted I believe the stone should be fine.

Comment (amative1) [reply src/scripts/pokemons/PokemonList.ts:15870]: Looking at the reqs on Bulbapedia, I'll be re-doing that portion later today. Closing the pull request for now, I'll resubmit when it's done.

**Pull request closed without merging**